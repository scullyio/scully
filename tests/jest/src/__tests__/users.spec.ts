import { readPage, replaceIndexNG } from '../test-config.helper';
import { expect } from '@jest/globals';

describe('JsonPlugin: test user List', () => {
  it('Check clean blog index by scully', () => {
    const index: string = readPage('/user');
    const cleanIndex = replaceIndexNG(index);
    expect(cleanIndex).toMatchSnapshot();
  });
});
