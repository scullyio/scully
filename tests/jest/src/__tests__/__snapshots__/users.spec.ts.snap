// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JsonPlugin: test user List Check clean blog index by scully 1`] = `
<!DOCTYPE html>
<html lang="en">
  <head>
    <meta name="generator">
    <meta charset="utf-8">
    <title>
      sample-blog
    </title>
    <base href="/">
    <meta name="viewport"
          content="width=device-width, initial-scale=1"
    >
    <link rel="icon"
          type="image/x-icon"
          href="favicon.ico"
    >
    <link rel="stylesheet"
          href="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.18.1/styles/default.min.css"
    >
    <link rel="stylesheet"
          href="styles.css"
    >
    <style>
      []{display:grid;height:100vh;grid-template-rows:60px 1fr 60px}[] > header[]{display:grid;background-color:#4169e1;color:#f5f5f5;margin:0;padding:0 10px;grid-template-columns:1fr 40px;place-items:center center}footer[]   h3[], header[]   h1[]{margin:0}[] > footer[]{display:grid;background-color:#4169e1;color:#f5f5f5;margin:0;padding-right:10px;justify-content:right;align-content:center}main[]{background-color:#f5f5f5;color:#040c24;padding:10px}
    </style>
    <style>
    </style>
  </head>
  <body>
    <app-root class="foo"
              =""
    >
      Scully demo blog app!
      <small ="">
        rendering inside scully
      </small>
      <a =""
         href="/home/"
      >
        üè†
      </a>
      <main ="">
        <router-outlet ="">
        </router-outlet>
        <app-users ="">
          <h1 ="">
            Users
          </h1>
          <section ="">
            <table ="">
              <thead ="">
                <tr>
                  <th ="">
                    Id
                  </th>
                  <th ="">
                    Username
                  </th>
                </tr>
              </thead>
              <tbody ="">
              </tbody>
            </table>
          </section>
        </app-users>
      </main>
      <footer ="">
        <h3 ="">
          Made with ‚ù§Ô∏è
          <strong ="">
            @HeroDevs
          </strong>
        </h3>
      </footer>
    </app-root>
  </body>
</html>
`;
